src/rgw/rgw_rest_s3.cc

```diff
@@ -4878,7 +4878,7 @@ RGWHandler_REST* RGWRESTMgr_S3::get_handler(rgw::sal::RGWRadosStore *store,
   } else {
     if (s->init_state.url_bucket.empty()) {
       handler = new RGWHandler_REST_Service_S3(auth_registry, enable_sts, enable_iam, enable_pubsub);
-    } else if (rgw::sal::RGWObject::empty(s->object.get())) {
+    } else if (rgw::sal::RGWObject::empty(s->object.get()) && !s->info.args.exist_obj_excl_sub_resource()) {
```

Comment:
This is the right idea, I think, but I don't think we want to dispatch into the object handler if we have no object.  How about re-arranging it so it looks like this:

if (!object.empty()) {
    handler = object_hander;
} else if (exist_obj_excl_sub_resource()) {
    return NULL;
} else {
    handler = bucket_handler;
}